services/web/server/src/simcore_service_webserver/projects/_common/exceptions_handlers.py

```diff
@@ -0,0 +1,188 @@
+    ),
+    ProjectWalletPendingTransactionError: HttpErrorInfo(
+        status.HTTP_409_CONFLICT,
+        "Project has currently pending transactions. It is forbidden to change wallet.",
+    ),
+    ProjectInDebtCanNotChangeWalletError: HttpErrorInfo(
+        status.HTTP_402_PAYMENT_REQUIRED,
+        "Unable to change the credit account linked to the project. The project is embargoed because the last transaction of {debt_amount} resulted in the credit account going negative.",
+    ),
+    ProjectInDebtCanNotOpenError: HttpErrorInfo(
+        status.HTTP_402_PAYMENT_REQUIRED,
+        "Unable to open the project. The project is embargoed because the last transaction of {debt_amount} resulted in the credit account going negative.",
+    ),
+    WrongTagIdsInQueryError: HttpErrorInfo(
+        status.HTTP_400_BAD_REQUEST,
+        "Wrong tag IDs in query",
+    ),
+}
+
+
+_WORKSPACE_ERRORS: ExceptionToHttpErrorMap = {
+    WorkspaceAccessForbiddenError: HttpErrorInfo(
+        status.HTTP_403_FORBIDDEN,
+        "Access to workspace forbidden: {reason}",
+    ),
+    WorkspaceNotFoundError: HttpErrorInfo(
+        status.HTTP_404_NOT_FOUND,
+        "Workspace not found: {reason}",
+    ),
+}
+
+
+_WALLET_ERRORS: ExceptionToHttpErrorMap = {
+    UserDefaultWalletNotFoundError: HttpErrorInfo(
+        status.HTTP_404_NOT_FOUND,
+        "Wallet not found: {reason}",
+    ),
+    WalletAccessForbiddenError: HttpErrorInfo(
+        status.HTTP_403_FORBIDDEN,
+        "Payment required, but the user lacks access to the project's linked wallet: Wallet access forbidden. {reason}",
+    ),
+    WalletNotEnoughCreditsError: HttpErrorInfo(
+        status.HTTP_402_PAYMENT_REQUIRED,
+        "Wallet does not have enough credits. {reason}",
+    ),
+}
+
+
+_PRICING_ERRORS: ExceptionToHttpErrorMap = {
+    DefaultPricingPlanNotFoundError: HttpErrorInfo(
+        status.HTTP_404_NOT_FOUND,
+        "Default pricing plan not found",
+    ),
+    DefaultPricingUnitNotFoundError: HttpErrorInfo(
+        status.HTTP_404_NOT_FOUND,
+        "Default pricing unit not found",
+    ),
+}
+
+
+_OTHER_ERRORS: ExceptionToHttpErrorMap = {
+    ClustersKeeperNotAvailableError: HttpErrorInfo(
+        status.HTTP_503_SERVICE_UNAVAILABLE,
+        "Clusters-keeper service is not available",
+    ),
+    CatalogForbiddenError: HttpErrorInfo(
+        status.HTTP_403_FORBIDDEN,
+        "Catalog forbidden: Insufficient access rights for {name}",
+    ),
+    CatalogItemNotFoundError: HttpErrorInfo(
+        status.HTTP_404_NOT_FOUND, "{name} was not found"
+    ),
+}
+
+_TO_HTTP_ERROR_MAP: ExceptionToHttpErrorMap = {}
+_TO_HTTP_ERROR_MAP.update(_FOLDER_ERRORS)
+_TO_HTTP_ERROR_MAP.update(_NODE_ERRORS)
+_TO_HTTP_ERROR_MAP.update(_PROJECT_ERRORS)
+_TO_HTTP_ERROR_MAP.update(_WORKSPACE_ERRORS)
+_TO_HTTP_ERROR_MAP.update(_WALLET_ERRORS)
+_TO_HTTP_ERROR_MAP.update(_PRICING_ERRORS)
+_TO_HTTP_ERROR_MAP.update(_OTHER_ERRORS)
```

Comment:
Consider something like this to avoid accidental duplicate keys.

```suggestion
_ALL_HTTP_ERROR_MAPS: list[ExceptionToHttpErrorMap] = [
    _FOLDER_ERRORS,
    _NODE_ERRORS,
    _PROJECT_ERRORS,
    _WORKSPACE_ERRORS,
    _WALLET_ERRORS,
    _PRICING_ERRORS,
    _OTHER_ERRORS,
]
_TO_HTTP_ERROR_MAP: ExceptionToHttpErrorMap = {**e for e in _ALL_HTTP_ERROR_MAPS}
assert len(_TO_HTTP_ERROR_MAP) == sum(len(e) for e in _ALL_HTTP_ERROR_MAPS) # nosec
```